### PR TITLE
Add ocaml 4.03 to hhvm-third-party so we can depend on it in hhvm

### PR DESCRIPTION
**Pull request by aorenste**

### CODE REVIEW

**Comment by fredemmott** [comment]
<3

**Comment by fredemmott** [comment]
:popcorn:

**Comment by jopamer** [comment]
Awesome!

File: ocaml/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,102 @@
+set(OCAMLC_FOUND FALSE)
+
+find_program(OCAMLC_EXECUTABLE NAMES ocamlc.opt ocamlc)
+find_program(OCAMLOPT_EXECUTABLE NAMES ocamlopt.opt ocamlopt)
+find_program(OCAMLBUILD_EXECUTABLE ocamlbuild.native ocamlbuild)
```
Comment (paulbiss) [inline ocaml/CMakeLists.txt]: NAMES?

Comment (aorenste) [inline ocaml/CMakeLists.txt]: Oops.

**Comment by aorenste** [comment]
Okay - updated to fix the issue Paul found.

**Comment by aorenste** [comment]
Updating again - @Orvid wants me to change it so ocaml is built at make time instead of cmake time.

**Comment by aorenste** [comment]
@Orvid Updated to build ocaml at make time.  Please review.

**Pull request merged**